datalad/support/archive_utils_7z.py

```diff
@@ -52,13 +55,14 @@ def decompress_file(archive, dir_):
     if len(suffixes) > 1 and suffixes[-2] == '.tar':
         # we have a compressed tar file that needs to be fed through the
         # decompressor first
-        # hangs somehow, do via single string arg
-        #cmd = ['7z', 'x', archive, '-so', '|', '7z', 'x', '-si', '-ttar']
-        cmd = '7z x {} -so | 7z x -si -ttar'.format(quote_cmdlinearg(archive))
+        cmd = [
+            'sh', '-c',
+            '7z x {} -so | 7z x -si -ttar'.format(quote_cmdlinearg(archive)),
+        ]
     else:
         # fire and forget
         cmd = ['7z', 'x', archive]
-    runner.run(cmd)
+    runner.run(cmd, protocol=StdOutErrCapture)
```

Comment:
Given that this function and the one below aren't returning the output, would it be better to use the `KillOutput` protocol?

Reply:
Oh. You are absolutely correct!

Reply:
Done